queries.sql

```diff
@@ -1,3 +1,244 @@
+and owners.id=animals.owner_id
+
+-- List of all animals that are pokemon (their type is Pokemon).
+
+select animals.name animals , species.name species from animals, species
+where species.id=animals.species_id
+and species.name='Pokemon'
+
+-- List all owners and their animals, remember to include those that don't own any animal.
+
+select animals.name animals , owners.full_name owners from animals right outer join owners 
+on owners.id=animals.owner_id
+
+-- How many animals are there per species?
+select count(animals.name) total_animals ,species.name species from animals inner join species
+on species.id=animals.species_id
+Group by species.name
+
+-- List all Digimon owned by Jennifer Orwell.
+select animals.name animals , owners.full_name owners, species.name from animals inner join owners
+on owners.id=animals.owner_id
+inner join species on
+species.id=animals.species_id
+where  species.name='Digimon'
+and owners.full_name='Jennifer Orwell' 
+
+-- List all animals owned by Dean Winchester that haven't tried to escape.
+select owners.full_name, animals.name , animals.escape_attempts from animals inner join owners
+on owners.id=animals.owner_id
+where owners.full_name='Dean Winchester'
+and animals.escape_attempts=0;
+select * from animals
+
+-- Who owns the most animals?
+
+select  count(animals.name) total_animals, owners.full_name from animals inner join owners
+on owners.id= animals.owner_id
+group by owners.full_name
+order by total_animals desc
+limit 1
+
+
+
+-- queries for many to many relations
+
+-- Who was the last animal seen by William Tatcher?
+
+select animals.name, vets.name, visits.date_of_vists from animals inner join visits on animals.id=visits.animal_id
+inner join vets on visits.vets_id=vets.id
+where vets.name='William Tatcher'
+order by visits.date_of_vists desc
+limit 1
+
+-- How many different animals did Stephanie Mendez see?
+
+select count(animals.name), vets.name from animals inner join visits on animals.id=visits.animal_id
+inner join vets on visits.vets_id=vets.id
+where vets.name='Stephanie Mendez'
+group by vets.name
+
+
+-- List all vets and their specialties, including vets with no specialties.
+
+select vets.name, species.name
+from specializations inner join species on specializations.species_id = species.id
+right outer join vets on specializations.vets_id = vets.id
+
+-- List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
+
+select animals.name, vets.name, visits.date_of_vists from animals inner join visits on animals.id=visits.animal_id
+inner join vets on visits.vets_id=vets.id
+where vets.name='Stephanie Mendez'
+and visits.date_of_vists BETWEEN '01-04-2020' AND '08-30-2020'
+
+-- What animal has the most visits to vets?
+Select count(animals.name), vets.name from animals inner join visits on animals.id=visits.animal_id
+inner join vets on vets.id= visits.vets_id
+group by vets.name
+order by count(animals.name) desc
+limit 1
```

Comment:
- [ ] in this query you are supposed to return the name of an animal whit the most visits to a vet, now it returns a count of animals who visited the vet with the most visits